File: benchmarks/python/fib.py
#!/usr/bin/env python3

TARGET = 28
REPEAT = 5
ANSWER = 317_811


def fib(n: int) -> int:
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)


result = 0
for _ in range(REPEAT):
    result = fib(TARGET)

if result != ANSWER:
    raise SystemExit(f"wrong answer: expected {ANSWER}, got {result}")

File: benchmarks/node/fib.js
const TARGET = 28;
const REPEAT = 5;
const ANSWER = 317811;

function fib(n) {
  if (n < 2) {
    return n;
  }
  return fib(n - 1) + fib(n - 2);
}

let result = 0;
for (let i = 0; i < REPEAT; i += 1) {
  result = fib(TARGET);
}

if (result !== ANSWER) {
  throw new Error(`wrong answer: expected ${ANSWER}, got ${result}`);
}
